### PR TITLE
Added Target JHEMU JHEF405A AIO with build in ELRS Serial RX, Baro and BMI IMU

### PR DESCRIPTION
**Pull request by ltwin8**
target for the GSF405A-BMI AIO from jhemcu

tested with GPS, DSHOT600, MAG, BARO and ELRS, working with no issues.

### CODE REVIEW

**Comment by scottrea123** [comment]
Checked this build with the JHEF405PRO and it does not look to be working correctly. 

Perhaps this target need to be renamed JHEF405A to avoid confusion?

**Comment by ltwin8** [comment]
I have flown it today, works without issueVon meinem iPhone gesendetAm 23.01.2023 um 20:40 schrieb scottrea123 ***@***.***>:﻿
Checked this build with the JHEF405PRO and it does not look to be working correctly.
Perhaps this target need to be renamed JHEF405A to avoid confusion?

**Comment by scottrea123** [comment]
You are using a GSF405A-BMI ?

So maybe name it that - JHEF405PRO is a different board.

**Comment by ltwin8** [comment]
The GSF405A-BMI uses the JHEF405PRO target.

**Comment by ltwin8** [comment]
[inav_5.1.0_JHEF405PRO.hex.zip](https://github.com/iNavFlight/inav/files/10484158/inav_5.1.0_JHEF405PRO.hex.zip)
could you try this binary? I have added the mpu6000 as that's maybe the issue, if your board does not use the bmi270

**Comment by ltwin8** [comment]
> You are using a GSF405A-BMI ?
> 
> So maybe name it that - JHEF405PRO is a different board.

could you maybe again test the hex file provided? if there is no problem there is no need to keep the board from official support since all i can say is the board works without any problem discovered (other than a bit of toiletbowling because of Magnetometer placement) since 22nd of january

**Comment by salukikev** [comment]
I have this board on order in hopes of using it with INAV.   It looks like these examples are for 5.1, will this target also work with the recently released INAV 6?

**Comment by salukikev** [comment]
I loaded the binary successfully, but I have two outstanding problems:
1. My Baro doesn't work.
2. I can't seem to correctly switch motor outputs around (using the mixer) as settings won't save. 
I plan to just rewire my motors and apply some awkward wire routing to remedy the 2nd one but it sure would be nice to get this onto INAV 6 because it is otherwise a great FC.

**Comment by ltwin8** [comment]
I have remapped the motors in the mixer and it works. Which baro does your FC use? Maybe you have to enable it? There isn’t much that can go wrong, since hard wired….Von meinem iPhone gesendetAm 26.04.2023 um 17:03 schrieb salukikev ***@***.***>:﻿
I loaded the binary successfully, but I have two outstanding problems:

My Baro doesn't work.
I can't seem to correctly switch motor outputs around (using the mixer) as settings won't save.
I plan to just rewire my motors and apply some awkward wire routing to remedy the 2nd one but it sure would be nice to get this onto INAV 6 because it is otherwise a great FC.

**Comment by salukikev** [comment]
> I have remapped the motors in the mixer and it works. Which baro does your FC use? Maybe you have to enable it? There isn’t much that can go wrong, since hard wired….Von meinem iPhone gesendetAm 26.04.2023 um 17:03 schrieb salukikev ***@***.***>:﻿ I loaded the binary successfully, but I have two outstanding problems: My Baro doesn't work. I can't seem to correctly switch motor outputs around (using the mixer) as settings won't save. I plan to just rewire my motors and apply some awkward wire routing to remedy the 2nd one but it sure would be nice to get this onto INAV 6 because it is otherwise a great FC. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

It claims to use BMP280, but that is from the aliexpress description, and I have tried each option and none produce a result.  The point may be moot for now as I have managed to brick this FC just when it was getting fun.  There doesn't seem to be any physical or electrical damage but nothing I've tried will get this into DFU mode. and window's won't recognize it anymore.  This is all the result of trying to load a questionably named alternative target from the vendors github.  Lesson learned.  I was trying to switch back to Betaflight to try to get more info about the baro at the time and determine if it was working there.

**Comment by Magicsmoke-mit-Lothar** [comment]
On the oficcial website https://www.jhemcu.com/e_productshow/?75-GSF405A-BMI-75.html it says it has the SPL06 barometer

**Comment by ltwin8** [comment]
Am 24.07.2023 um 21:07 schrieb Magicsmoke_mit_Lothar ***@***.***>:﻿
On the oficcial website https://www.jhemcu.com/e_productshow/?75-GSF405A-BMI-75.html it says it has the SPL06 barometer

**Comment by ltwin8** [comment]
Just looked again, seems they changed without notification.

**Comment by mxchinist** [comment]
I've fixed this by adding: #define USE_BARO_DSP310 under the #define USE_BARO_BMP280 line in target.h file before building the target. (Also I have removed the USE_BARO_MS5611 line )
Baro icon in inav is now blue.

File: src/main/target/JHEF405PRO/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+target_stm32f405xg(JHEF405PRO)
```
Comment (mmosca) [inline src/main/target/JHEF405PRO/CMakeLists.txt:1]: Can you add SKIP_RELEASES ?

**Comment by mitsu199** [comment]
Nice to meet you, everyone.
Please help me with GSF405A-BMI AIO from jhemcu barometer.
I got inav6 for gsf405a-bmi from github, jhemcu and flashed it to the board.
The barometer is blue in the inav configurator, but the barometer display is negative when hovering.
Inaccurate numbers such as -20, -30 are displayed when hovering about 2 meters.
So I flashed Betaflight and checked and the barometer showed normal readings.
The barometer was not broken.
What causes inav6 to not work properly?
I can't solve the problem any more.
Please help me. please tell me.
It's a google translation, so it might be hard to read.
sorry.

**Comment by mitsu199** [comment]
> I've fixed this by adding: #define USE_BARO_DSP310 under the #define USE_BARO_BMP280 line in target.h file before building the target. (Also I have removed the USE_BARO_MS5611 line ) Baro icon in inav is now blue.

nice to meet you.
Mr. mxchinist.
Could you share the inav you built with me?
I am unable to build.
Please.

File: src/main/target/JHEF405PRO/target.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#include <stdbool.h>
+#include <platform.h>
+#include "drivers/io.h"
+#include "drivers/pwm_mapping.h"
+#include "drivers/timer.h"
+
+timerHardware_t timerHardware[] = {
+    DEF_TIM(TIM4, CH3, PB8, TIM_USE_PPM, 0, 0), // PPM
+    DEF_TIM(TIM3, CH3, PB0, TIM_USE_MC_MOTOR, 0, 0),  // S1
```
Comment (mmosca) [inline src/main/target/JHEF405PRO/target.c:26]: Can you change these to TIM_USE_MOTOR?

**Comment by SebastianKumor** [comment]
So is this board supported or not ? I cannot see the JHEF405PRO target in 7.1 anymore

**Comment by SebastianKumor** [comment]
@mmosca I would take over if @ltwin8  abandoned this

**Comment by ltwin8** [comment]
I have a working fork and installed inav on that board. It’s on my profile.

**Comment by SebastianKumor** [comment]
@ltwin8 why not merge this this then

**Comment by mmosca** [comment]
Official targets added

**Pull request closed without merging**